src/SUMMARY.md

```diff
@@ -63,6 +63,8 @@
         - [Match expressions](expressions/match-expr.md)
         - [Return expressions](expressions/return-expr.md)
 
+- [Patterns](patterns.md)
```

Comment:
Let me know if you want to move the patterns chapter elsewhere (such as underneath `Statements and Expressions`).

Reply:
I want to eventually to split all the "X and Y" chapters eventually, so having this top-level is good to me.

Reply:
To expand on this, I'd like to see the order be "Items, Attributes, Expressions, Patterns, Statements", so the current location LGTM as well.